apps/src/templates/teacherDashboard/teacherSectionsRedux.js

```diff
@@ -398,6 +400,13 @@ export default function teacherSections(state=initialState, action) {
     };
   }
 
+  if (action.type === SET_CSF_SCRIPT_IDS) {
+    return {
+      ...state,
+      isCsfScript: id => action.ids.indexOf(id) > -1,
```

Comment:
Code smell: Functions in Redux state.

[From the Redux FAQ](https://redux.js.org/faq/organizing-state#can-i-put-functions-promises-or-other-non-serializable-items-in-my-store-state):

> **Can I put functions, promises, or other non-serializable items in my store state?**
> It is highly recommended that you only put plain serializable objects, arrays, and primitives into your store. It's technically possible to insert non-serializable items into the store, but doing so can break the ability to persist and rehydrate the contents of a store, as well as interfere with time-travel debugging.
>
> If you are okay with things like persistence and time-travel debugging potentially not working as intended, then you are totally welcome to put non-serializable items into your Redux store.  Ultimately, it's your application, and how you implement it is up to you.  As with many other things about Redux, just be sure you understand what tradeoffs are involved.

Obviously we're not using persistence or time-travel debugging in our application, but I worry that patterns like this are likely to interfere with upgrades down the road - either newer versions of Redux that enforce stricter rules for the sake of performance optimizations as we've seen a few times with React, or with future attempts by our team to introduce some related tool (like prehydrating a redux store on the server, for instance) that would require stricter guidelines.

Instead, I would export a specialized selector function and call it in your `mapStateToProps` function when connecting `EditSectionForm`.

```js
// teacherSectionsRedux.js
export const isCsfScript = (state, id) => state.csfScriptIds.indexOf(id) > -1;

// EditSectionForm.jsx
export default connect(state => ({
  isCsfScript: id => isCsfScript(state, id)
}))
```

Reply:
Good catch, updated in a1ab5d2fdf924d3a834ff862f9a988633dd407cf.